src/table/src/predicate/stats.rs

```diff
@@ -40,92 +37,59 @@ impl<'a> RowGroupPruningStatistics<'a> {
-                                Some(ScalarValue::Float64($min_max))
-                            }
-                            PhysicalType::ByteArray => {
-                                let $min_max = stats
-                                    .as_any()
-                                    .downcast_ref::<BinaryStatistics>()
-                                    .unwrap()
-                                    .[<$min_max _value>]
-                                    .clone();
-                                Some(ScalarValue::Binary($min_max))
-                            }
-                            PhysicalType::FixedLenByteArray(_) => {
-                                let $min_max = stats
-                                    .as_any()
-                                    .downcast_ref::<FixedLenStatistics>()
-                                    .unwrap()
-                                    .[<$min_max _value>]
-                                    .clone();
-                                Some(ScalarValue::Binary($min_max))
-                            }
-                        };
+    ($self:ident, $col:ident, $min_max: ident) => {{
+        let arrow_schema = $self.schema.arrow_schema().clone();
+        let (column_index, field) = if let Some((v, f)) = arrow_schema.column_with_name(&$col.name)
+        {
+            (v, f)
+        } else {
+            return None;
+        };
+        let data_type = field.data_type();
+        let null_scalar: ScalarValue = if let Ok(v) = data_type.try_into() {
+            v
+        } else {
+            return None;
+        };
 
-                        res
-                    })
-                    .map(|maybe_scalar| maybe_scalar.unwrap_or_else(|| null_scalar.clone()))
-                    .collect::<Vec<_>>();
-                ScalarValue::iter_to_array(scalar_values).ok()
-        }
-    }
-    };
+        let scalar_values = $self
+            .meta_data
+            .iter()
+            .flat_map(|meta| meta.column(column_index).statistics())
+            .map(|stats| {
+                if !stats.has_min_max_set() {
+                    return None;
+                }
+                match stats {
+                    ParquetStats::Boolean(s) => Some(ScalarValue::Boolean(Some(*s.$min_max()))),
+                    ParquetStats::Int32(s) => Some(ScalarValue::Int32(Some(*s.$min_max()))),
+                    ParquetStats::Int64(s) => Some(ScalarValue::Int64(Some(*s.$min_max()))),
+
+                    ParquetStats::Int96(_) => None,
+                    ParquetStats::Float(s) => Some(ScalarValue::Float32(Some(*s.$min_max()))),
+                    ParquetStats::Double(s) => Some(ScalarValue::Float64(Some(*s.$min_max()))),
+                    ParquetStats::ByteArray(s) => {
+                        paste! {
+                            let s = std::str::from_utf8(s.[<$min_max _bytes>]())
+                                .map(|s| s.to_string())
```

Comment:
Could we use [`String::from_utf8`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8)?
```suggestion
                            let s = String::from_utf8(s.[<$min_max _bytes>]())
```